go/src/lib.rs

```diff
@@ -42,14 +41,11 @@ pub struct CommandResponse {
     string_value: *mut c_char,
     string_value_len: c_long,
 
-    // Below three values are related to each other.
-    // `array_value` represents the array of strings.
-    // `array_elements_len` represents the length of each array element.
+    // Below two values are related to each other.
+    // `array_value` represents the array of CommandResponse.
     // `array_value_len` represents the length of the array.
     #[derivative(Default(value = "std::ptr::null_mut()"))]
-    array_value: *mut *mut c_char,
-    #[derivative(Default(value = "std::ptr::null_mut()"))]
-    array_elements_len: *mut c_long,
+    array_value: *mut CommandResponse,
     array_value_len: c_long,
```

Comment:
can we add the type enum / int right away?

Reply:
Added a TODO, will handle it in the upcoming PR.